railties/lib/rails/generators/test_unit/scaffold/scaffold_generator.rb

```diff
@@ -11,12 +11,19 @@ class ScaffoldGenerator < Base # :nodoc:
       class_option :api, type: :boolean,
                          desc: "Generates API functional tests"
 
+      class_option :system_tests, type: :string,
+                         desc: "Skip system test files"
+
       argument :attributes, type: :array, default: [], banner: "field:type field:type"
 
       def create_test_files
         template_file = options.api? ? "api_functional_test.rb" : "functional_test.rb"
         template template_file,
                  File.join("test/controllers", controller_class_path, "#{controller_file_name}_controller_test.rb")
+
+        unless options.api? || options[:system_tests].nil?
+          template "system_test.rb", File.join("test/system", class_path, "#{file_name.pluralize}_test.rb")
+        end
```

Comment:
You're right, it's not supported on master. I think this PR is fine then and if we want to implement `--skip-system-test` for scaffold we can do that in a separate PR.